lifetimes/estimation.py

```diff
@@ -37,6 +38,117 @@ def _print_params(self):
         return s.strip(', ')
 
 
+class GammaGammaFitter(BaseFitter):
+    def __init__(self, penalizer_coef=0.):
+        self.penalizer_coef = penalizer_coef
+
+    @staticmethod
+    def _negative_log_likelihood(params, frequency, avg_monetary_value, penalizer_coef):
+        if any(i < 0 for i in params):
+            return np.inf
+
+        p, q, v = params
+
+        x = frequency.astype('int')
+        m = avg_monetary_value.astype('int')
+
+        p_0 = special.gamma((p * x) + q) / (special.gamma(p * x) * special.gamma(q))
+        p_1 = ((v ** q) + (m ** ((p * x) - 1)) * (x ** (p * x))) / ((v + (m * x)) ** ((p * x) + q))
+
+        negative_log_likelihood_values = (-np.log(p_0 * p_1))
+        negative_log_likelihood_values = negative_log_likelihood_values.dropna()
+        negative_log_likelihood_values = negative_log_likelihood_values[
+            np.abs(negative_log_likelihood_values) != np.inf]
+
+        penalizer_term = penalizer_coef * log(params).sum()
+        negative_log_likelihood = np.sum(negative_log_likelihood_values) + penalizer_term
+
+        if np.any(np.isnan(negative_log_likelihood_values)) or np.isnan(negative_log_likelihood):
+            return np.inf
+
+        return negative_log_likelihood
+
+    def conditional_expected_average_profit(self, x=None, m=None):
+        """
+        This method computes the conditional expectation of the average profit per transaction
+        for a group of one or more customers.
+            x: a vector containing the customers' frequencies. Defaults to the whole set of
+                frequencies used for fitting the model.
+            m: a vector containing the customers' monetary values. Defaults to the whole set of
+                monetary values used for fitting the model.
+
+        Returns:
+            the conditional expectation of the average profit per transaction
+        """
+        m = self.data['monetary_value'] if m is None else m
+        x = self.data['frequency'] if x is None else x
+        p, q, v = self.params_.values()
+        return np.mean((((q - 1) / (p * x + q - 1)) * (v * p / (q - 1))) + (p * x / (p * x + q - 1)) * m)
+
+    def customer_lifetime_value(self, transaction_prediction_model, frequency, recency, T, monetary_value, time=12, discount_rate=1):
```

Comment:
m and x can be vectors containing 1..n values. so this is for any amount of customers. should I make it more clear?